python/StructuredDialects.cpp

```diff
@@ -6,19 +6,35 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include <mlir-c/BuiltinAttributes.h>
 #include <pybind11/pybind11.h>
 
-#include <array>
-#include <initializer_list>
+#include <string>
 
-#include "mlir-c/BuiltinAttributes.h"
 #include "mlir-c/IR.h"
 #include "mlir/Bindings/Python/PybindAdaptors.h"
 #include "structured-c/Dialects.h"
+#include <mlir-c/BuiltinTypes.h>
+#include <vector>
 
 namespace py = pybind11;
 using namespace mlir::python::adaptors;
 
+static MlirStringRef toMlirStringRef(const std::string &s) {
+  return mlirStringRefCreate(s.data(), s.size());
+}
+
+namespace {
+// TODO(max): this is plucked from PybindUtils.cpp upstream - refactor upstream
+pybind11::error_already_set SetPyError(PyObject *excClass,
+                                       const llvm::Twine &message) {
+  auto messageStr = message.str();
+  PyErr_SetString(excClass, messageStr.c_str());
+  return pybind11::error_already_set();
+}
```

Comment:
This function isn't used (it was during experimentation) but it couldn't be pulled out of PybindUtils.cpp because PybindUtils.cpp is not exported/exposed because the [directory it lives in (as well as its header)](https://github.com/llvm/llvm-project/tree/main/mlir/lib/Bindings/Python) technically isn't "includable" (you can play [tricks](https://github.com/makslevental/nelli/blob/2dc46b3655b5aa7f4ba4816cc094c7b7ee1bb644/cpp_ext/CMakeLists.txt#L70)). I've discussed with @ftynse moving various things out of that directory but we just haven't come to a consensus yet on what and where.

Reply:
Ah, OK, got it. I knew this kind of problem and I *thought* I had checked whether it was the case here but apparently came to the wrong conclusion...